lib/pkgcloud/openstack/cdn/client/services.js

```diff
@@ -0,0 +1,311 @@
+/*
+ * services.js: Instance methods for working with services from Openstack CDN
+ *
+ * (C) 2014 Rackspace
+ *      Shaunak Kashyap
+ * MIT LICENSE
+ */
+
+var pkgcloud = require('../../../../../lib/pkgcloud'),
+  errs = require('errs'),
+  url = require('url'),
+  urlJoin = require('url-join'),
+  util = require('util'),
+  _ = require('underscore'),
+  cdn = pkgcloud.providers.openstack.cdn,
+  jsonpatch = require('fast-json-patch');
+
+// Declaring variables for helper functions defined later
+var _getService;
+
+var _urlPrefix = '/services';
+
+/**
+ * validateProperties
+ *
+ * @description local helper function for validating arguments
+ *
+ * @param {Array}       required      The list of required properties
+ * @param {object}      options       The options object to validate
+ * @param {String}      formatString  String formatter for the error message
+ * @param {Function}    callback
+ * @returns {boolean}
+ */
+function validateProperties(required, options, formatString, callback) {
```

Comment:
Tradition mostly :smile:, based on other services' code. I'll move it to the bottom.